AmplifyPlugins/Core/AWSPluginsCore/WebSocket/AmplifyNetworkMonitor.swift

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+import Network
+import Combine
+
+@_spi(WebSocket)
+public final class AmplifyNetworkMonitor {
+
+    public enum State {
+        case none
+        case online
+        case offline
+    }
+
+    #if DEBUG
+    private let monitor = NWPathMonitor(requiredInterfaceType: .cellular)
```

Comment:
leaving a comment here so we don't lose track. I don't think we should be setting only `celluar` unless this was a specific way to test this in DEBUG mode. But removing the hardcode means we should provide a way to test it, such as being able to pass `.cellular` to AmplifyNetworkMonitor for testing. Can we pass it into the `init` as an `AmplifyNetworkMonitor. InterfaceType` ?

Reply:
👍 will add the network interfaces in the injection way.